src/Accounts/Authentication/Config/ConfigInitializer.cs

```diff
@@ -0,0 +1,156 @@
+                {
+                    return path;
+                }
+            }
+            // if not found, use the first writable path
+            foreach (string path in paths)
+            {
+                try
+                {
+                    DirectoryInfo dir = new FileInfo(path).Directory;
+                    DataStore.CreateDirectory(dir.FullName); // create directory if not exists
+                    using (var _ = DataStore.OpenForExclusiveWrite(path)) { }
+                    return path;
+                }
+                catch (Exception)
+                {
+                    continue;
+                }
+            }
+            throw new ApplicationException($"Failed to store the config file. Please make sure any one of the following paths is accessible: {string.Join(", ", paths)}");
+        }
+
+        internal IConfigManager GetConfigManager()
+        {
+            lock (_fsLock)
+            {
+                ValidateConfigFile();
+            }
+            return new ConfigManager(_pathToConfigFile, DataStore, EnvironmentVariableProvider);
+        }
+
+        private void ValidateConfigFileContent()
+        {
+            string json = DataStore.ReadFileAsText(_pathToConfigFile);
+
+            bool isValidJson = true;
+            try
+            {
+                JObject.Parse(json);
+            }
+            catch (Exception)
+            {
+                isValidJson = false;
+            }
+
+            if (string.IsNullOrEmpty(json) || !isValidJson)
+            {
+                Debug.Write($"[ConfigInitializer] Failed to parse the config file at {_pathToConfigFile}. Clearing the file.");
+                ResetConfigFileToDefault();
+            }
+        }
+
+        private void ValidateConfigFile()
+        {
+            if (!DataStore.FileExists(_pathToConfigFile))
+            {
+                ResetConfigFileToDefault();
+            }
+            else
+            {
+                ValidateConfigFileContent();
+            }
+        }
+
+        private void ResetConfigFileToDefault()
+        {
+            try
+            {
+                DataStore.WriteFile(_pathToConfigFile, @"{}");
+            }
+            catch (Exception ex)
+            {
+                // do not halt for IO exception
+                Debug.WriteLine(ex.Message);
+            }
+        }
+
+        // todo: tests initializes configs in a different way. Maybe there should be an abstraction IConfigInitializer and two concrete classes ConfigInitializer + TestConfigInitializer
+        internal void InitializeForAzureSession(AzureSession session)
+        {
+            IConfigManager configManager = GetConfigManager();
+            session.RegisterComponent(nameof(IConfigManager), () => configManager);
+            RegisterConfigs(configManager);
+            configManager.BuildConfig();
+        }
+
+        private void RegisterConfigs(IConfigManager configManager)
```

Comment:
This method will be implemented in a follow up PR

Reply:
It is great that those code can be removed.

Reply:
removed